gravitee-alert-engine-connector-ws/src/main/java/io/gravitee/ae/connector/ws/WebSocketConnector.java

```diff
@@ -193,6 +196,29 @@ private void doConnect(Promise<WebSocket> webSocketPromise) {
             webSocketConnectOptions.setHeaders(getDefaultHeaders(engine));
             webSocketConnectOptions.setURI(target.getRawPath() + path);
 
+            var connectorConfig = engine.getConnectorConfiguration();
+
+            if (connectorConfig.isUseSystemProxy()) {
+                ProxyOptions proxyOptions = new ProxyOptions().setType(ProxyType.valueOf(connectorConfig.getProxyType()));
+                if (HTTPS_SCHEME.equals(target.getScheme())) {
+                    webSocketConnectOptions.setProxyOptions(
+                        proxyOptions
+                            .setHost(connectorConfig.getProxyHttpsHost())
+                            .setPort(connectorConfig.getProxyHttpsPort())
+                            .setUsername(connectorConfig.getProxyHttpsUsername())
+                            .setPassword(connectorConfig.getProxyHttpsPassword())
+                    );
+                } else {
+                    webSocketConnectOptions.setProxyOptions(
+                        proxyOptions
+                            .setHost(connectorConfig.getProxyHttpHost())
+                            .setPort(connectorConfig.getProxyHttpPort())
+                            .setUsername(connectorConfig.getProxyHttpUsername())
+                            .setPassword(connectorConfig.getProxyHttpPassword())
+                    );
+                }
+            }
```

Comment:
This code seems duplicate with one in `Engine.java`. However, I don't find any common between `WebSocketConnectOptions` and `HttpClientOptions`. Both have `setProxyOptions` but not compatible. So, I think that the duplicate of code here is better and simpler ^^